### PR TITLE
fix grafana routes e2e test - list secrets

### PR DESCRIPTION
**Pull request by jhellar**
# Description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: test/common/grafana_routes.go [hunk 1]
```diff
@@ -95,38 +95,40 @@ func TestGrafanaExternalRouteDashboardExist(t *testing.T, ctx *TestingContext) {
...
-	for _, secret := range serviceAccount.Secrets {
-		if strings.Contains(secret.Name, "token") {
-			secretName = secret.Name
+	token := ""
+	secrets, err := ctx.KubeClient.CoreV1().Secrets(grafanaNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatal("failed to get secrets", err)
+	}
+	for _, secretsItem := range secrets.Items {
+		if strings.HasPrefix(secretsItem.Name, "test-token-") {
```
Comment (david-martin) [inline test/common/grafana_routes.go:110]: @jhellar will there only ever be 1 with this prefix?

Comment (jhellar) [reply test/common/grafana_routes.go:110]: There are 2 secrets created with this prefix when the service account is created. One of them can be used as the bearer token. By looking at both of them, the only difference I could found was `kubernetes.io/created-by` annotation, that is not present in the secret with the bearer token [1].

[1] https://github.com/integr8ly/integreatly-operator/pull/1150/files#diff-ab3b7c7cf278b4574268aa4ce059a50aR122

**Pull request merged**